### PR TITLE
Feature/no web3modal

### PR DESCRIPTION
**Pull request by PorlyBe**
**Short Description:**

- dont use web3 modal
- create custom wallletconnect connector for wagmi (using legacy "chains" option)
- create "connect" modal with walletconnect and injected provider
- create game mode selector
- update amplitude events

**Figma Link (optional):**
N/A

**Screenshots (optional):**
image

image
image

**This PR is blocked by or a continuation of:**
PR#

### CODE REVIEW

File: src/components/Header/Connect/GameModesList.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+          >
+            <img src={game.icon} alt={game.id} height={24} width={24} />
+            <div
+              style={{
+                marginLeft: "0.5rem",
+                color: "inherit",
+                background: "inherit",
+              }}
+            >
+              {game.label} {game.id === "LunarNewYear" && `(${chain?.name})`}
```
Comment (allisonle) [inline src/components/Header/Connect/GameModesList.tsx:118]: Maybe exclude the chain name if user hasn't connected wallet yet?
Screenshot 2024-02-16 at 12 23 07 PM

File: src/components/Header/Connect/GameModesList.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      {isConnected && (
+        <Button
+          $size="small"
+          $design="secondary"
+          onClick={() => {
+            disconnect();
+          }}
+        >
+          Disconnect
+        </Button>
```
Comment (allisonle) [inline src/components/Header/Connect/GameModesList.tsx:133]: imo I would expect to see this in the modal I get when clicking on my wallet address, was a bit unexpected when I clicked on my address and was automatically disconnected 🤔

Not a blocker tho

File: src/components/Header/Connect/GameModesList.tsx (hunk 1 above)
Comment (PorlyBe) [reply src/components/Header/Connect/GameModesList.tsx:118]: oh good point 😄

File: src/components/Header/Connect/GameModesList.tsx (hunk 2 above)
Comment (PorlyBe) [reply src/components/Header/Connect/GameModesList.tsx:133]: this was me being lazy, lool... will update

**Review by allisonle — APPROVED** [review]
Would be nice to prettify the connect modal to match web3modal's in a separate commit, but otherwise looks good let it ripppp

**Pull request merged**